### PR TITLE
mastodon: 3.4.1 -> 3.4.4

### PR DESCRIPTION
**Pull request by erictapen**
[Release Notes 3.4.2](https://github.com/mastodon/mastodon/releases/tag/v3.4.2)
[Release Notes 3.4.3](https://github.com/mastodon/mastodon/releases/tag/v3.4.3)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
It is possible to repackage json packages using the new method - https://github.com/NixOS/nixpkgs/pull/140701 ?

**Comment by Izorkin** [comment]
Mastodon updated to v3.4.3.

**Comment by erictapen** [comment]
@SuperSandro2000 You offer the possibility to ping you on your GitHub profile. Wouly you mind taking a look on this?

File: pkgs/servers/mastodon/source.nix [hunk 1]
```diff
@@ -2,8 +2,8 @@
 { fetchgit, applyPatches }: let
   src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/servers/mastodon/source.nix:3]: nit: this should use fetchFromGitHub which should download a archive which is faster and more reproducible.

**Pull request merged**